tools/codegen/gen.py

```diff
@@ -646,25 +657,25 @@ def compute_declaration_yaml(f: NativeFunction) -> object:
         ('device_guard', f.device_guard),
         ('with_gil', False),
         ('deprecated', False),
-        ('has_math_kernel', DispatchKey.CompositeImplicitAutograd in f.dispatch),
+        ('has_math_kernel', f.has_composite_implicit_autograd_kernel),
     ])
 
 # See Note [Auto generated composite kernels]
 def has_autogenerated_composite_kernel(f: NativeFunction) -> bool:
     return (f.structured or f.structured_delegate is not None) and \
         (f.func.kind() == SchemaKind.functional or f.func.kind() == SchemaKind.inplace)
 
-@with_native_function
-def compute_registration_declarations(f: NativeFunction) -> str:
+@with_native_function_and_indices
+def compute_registration_declarations(f: NativeFunction, backend_indices: Dict[DispatchKey, BackendIndex]) -> str:
     name = dispatcher.name(f.func)
     returns_type = dispatcher.returns_type(f.func.returns).cpp_type_registration_declarations()
     args = dispatcher.arguments(f.func)
     args_str = ', '.join(a.no_default().decl_registration_declarations() for a in args)
     comment_data : Dict[str, str] = {
         'schema': f'aten::{f.func}',
         # TODO: What exactly is the semantics of the 'dispatch' field?
-        'dispatch': str(f.dispatch.keys() != {DispatchKey.CompositeImplicitAutograd}),
-        'default': str(any(is_generic_dispatch_key(k) for k in f.dispatch) or has_autogenerated_composite_kernel(f))
+        'dispatch': str({k for k, v in backend_indices.items() if v.has_backend(f)} != {DispatchKey.CompositeImplicitAutograd}),
```

Comment:
Probably need to figure out if we can just kill this field LOL

Reply:
i just figured that would be part of the grander "kill RegistrationDeclarations.yaml" project 😛